src/strings.bif

```diff
@@ -1556,39 +1572,24 @@ function remove_prefix%(str: string, sub: string%) : string
 	// This could just use repeated calls to lstrip(), except for a couple of reasons:
 	// 1) lstrip() creates a StringVal at the end, and that would mean repeated recreation of objects
 	// 2) lstrip() searches for any character in the string, not the string as a whole.
-	string s = str->ToStdString();
-	string sub_s = sub->ToStdString();
+	auto s = str->ToStdStringView();
+	auto sub_s = sub->ToStdStringView();
 
-	size_t pos = s.find(sub_s);
-	if ( pos != 0 )
-		return zeek::IntrusivePtr<zeek::StringVal>(NewRef{}, str);
```

Comment:
That optimization (not allocating a new string when the old one doesn't start with prefix is gone now?)

Not quite sure why it's otherwise 8% slower in the microbenchmark (and might not matter in the grand scheme), possibly the function calls and repeated string_view creation?